### PR TITLE
fix(websocket): Make websocket reconnect more reliably

### PR DESCRIPTION
**Pull request by KarlMae**
## Description of the change

Make websocket reconnect more reliably

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (Changed implementation, but existing functionality and APIs are not changed)

## Related issues

https://github.com/Qminder/server/issues/18383

### CODE REVIEW

File: test/unit/GraphQLSubscriptions.test.ts [hunk 1]
```diff
@@ -38,28 +41,18 @@ describe('GraphQL subscriptions', () => {
...
       );
-      expect(spy).toBeCalledWith(
+      expect(WebSocket).toBeCalledWith(
         `wss://api.qminder.com:443/graphql/subscription?rest-api-key=${keyValue}`,
       );
     });
   });
 
   describe('with websocket cleanup', () => {
-    afterEach(async () => {
```
Comment (KarlMae) [inline test/unit/GraphQLSubscriptions.test.ts:55]: This block was testing the following line:

https://github.com/Qminder/javascript-api/pull/518/files#diff-b696092cf2ea982882e23e816d35f6a093ddf5f9bcee13768906934886fb41c8L304


I think it makes sense to detect connection hang-ups in a singular place and that will be ping-pongs and keepalives (at least for some time) so it made sense to not handle non-user-related hangup in the onclose event

File: src/services/GraphQLService.ts [hunk 2]
```diff
@@ -258,16 +248,21 @@ export class GraphQLService {
...
+        this.connectionStatus,
+      )
+    ) {
       return;
     }
-    this.setConnectionStatus(ConnectionStatus.CONNECTING);
-    this.fetchTemporaryApiKey().then((tempApiKey: string) => {
-      this.createSocketConnection(tempApiKey);
+
+    console.info('[Qminder API - 28.04.]: Trying to connect to websocket!');
```
Comment (2ndalpha) [inline src/services/GraphQLService.ts:259]: Remove

File: src/services/GraphQLService.ts [hunk 3]
```diff
@@ -277,58 +272,56 @@ export class GraphQLService {
...
+      const response = await this.fetch(
+        `https://${this.apiServer}/${url}`,
+        body,
+      );
+      const responseJson = await response.json();
+      return responseJson.key;
+    } catch (e) {
+      const timeOut = Math.min(60000, Math.max(5000, 2 ** retryCount * 1000));
+      console.warn(
+        `[Qminder API - 28.04.]: Failed fetching temporary API key! Retrying in ${
```
Comment (2ndalpha) [inline src/services/GraphQLService.ts:285]: Update the copy

**Review by 2ndalpha — CHANGES_REQUESTED** [review]

File: src/services/GraphQLService.ts [hunk 4]
```diff
@@ -214,7 +204,7 @@ export class GraphQLService {
    */
   openPendingWebSocket(): void {
```
Comment (stdevi) [inline src/services/GraphQLService.ts:205]: @KarlMae I wonder, why `openPendingWebSocket` is an unused method?

File: src/services/GraphQLService.ts [hunk 5]
```diff
@@ -277,58 +272,56 @@ export class GraphQLService {
...
-
-        console.log(
-          `[GraphQL subscription] Reconnecting in ${
-            newTimeout / 1000
-          } seconds...`,
-        );
-        this.retryTimeout = setTimeout(this.openSocket.bind(this), newTimeout);
-
-        this.connectionRetries += 1;
+      if (event.code === CLIENT_SIDE_CLOSE_EVENT) {
```
Comment (stdevi) [inline src/services/GraphQLService.ts:316]: I wonder, what if code is 1006?

**Review by 2ndalpha — APPROVED** [review]
🚀

**Pull request merged**